shared/src/data-access/accessors/mongodb.ts

```diff
@@ -170,6 +261,142 @@ export class MongoDBAccessor extends BaseDatabaseAccessor {
+      deleteOne: (f) => collection.deleteOne(f),
+      deleteMany: (f) => collection.deleteMany(f),
+    };
+
+    const executeOperation = operationHandlers[operation];
+
+    if (!executeOperation) {
+      throw new Error(`Unsupported operation: ${operation}`);
+    }
+
+    // Success checkers map
+    const hasResults = (result: any): boolean => {
+      if (operation.includes('update') || operation.includes('delete')) {
+        return result.matchedCount > 0 || result.deletedCount > 0;
+      }
+
+      return false;
+    };
+
+    // Try alternative ID formats
+    for (const alternativeId of alternativeFormats) {
+      const alternativeFilter = { ...filter, _id: alternativeId };
+
+      try {
+        const result = await executeOperation(alternativeFilter, updateOrOptions, options);
+
+        if (hasResults(result)) {
+          return result;
+        }
+      } catch {
+        // Continue to next format
+      }
+    }
+
+    // Return default "not found" result based on operation type
+    const defaultResults: { [key: string]: any } = {
+      updateOne: { matchedCount: 0, modifiedCount: 0, acknowledged: true, upsertedId: null },
+      updateMany: { matchedCount: 0, modifiedCount: 0, acknowledged: true, upsertedCount: 0, upsertedId: null },
+      deleteOne: { deletedCount: 0, acknowledged: true },
+      deleteMany: { deletedCount: 0, acknowledged: true },
+    };
+
+    return defaultResults[operation] || {};
+  }
+
+  private _getAlternativeIdFormats(originalId: any): any[] {
+    const alternatives = [];
+
+    // If it's a number, try string
+    if (typeof originalId === 'number') {
+      alternatives.push(originalId.toString());
+    }
+
+    // If it's a string that looks like a number, try number
+    if (typeof originalId === 'string' && /^\d+$/.test(originalId)) {
+      const numValue = parseInt(originalId, 10);
+
+      if (!isNaN(numValue)) {
+        alternatives.push(numValue);
+      }
+    }
+
+    // If it's a string that looks like an ObjectId, try ObjectId
+    if (typeof originalId === 'string' && /^[0-9a-fA-F]{24}$/.test(originalId)) {
+      try {
+        alternatives.push(new ObjectId(originalId));
+      } catch {
+        // Ignore conversion error
+      }
+    }
+
+    // If it's an ObjectId, try string
+    if (
+      originalId &&
+      typeof originalId === 'object' &&
+      originalId.constructor &&
+      originalId.constructor.name === 'ObjectId'
+    ) {
+      alternatives.push(originalId.toString());
+    }
+
+    return alternatives;
+  }
+
+  private _convertIdValue(value: any): any {
+    // If the value is already an ObjectId, keep it
+    if (value && typeof value === 'object' && value.constructor && value.constructor.name === 'ObjectId') {
+      return value;
+    }
+
+    // For string values that look like ObjectIds (24 hex characters), try to convert
+    if (typeof value === 'string' && /^[0-9a-fA-F]{24}$/.test(value)) {
```

Comment:
Ok so I'm guessing you pretty much had to write the "parse the MQL string to an object" in order to execute MongoDB queries that will live in the built app. 

My question was, do we have to do that by hand? 
Is there an existing alternative that can help out, like https://www.npmjs.com/package/mongodb-query-parser ? 

Thanks.